Fix for null HttpFilters instance

jekh opened the pull request:
The [javadoc for HttpFiltersSource](https://github.com/adamfisk/LittleProxy/blob/master/src/main/java/org/littleshoot/proxy/HttpFiltersSource.java#L20) states:
```
Return an {@link HttpFilters} object for this request if and only if we 
want to filter request and/or its responses.
```

The "if and only if" part implies that the HttpFiltersSource should not return an HttpFilters instance (i.e. should return null) if we do not want to filter this request/response. Currently, this behavior is broken, and causes NPEs.

In addition to the scenario where the HttpFiltersSource returns a null HttpFilters, the HttpFilters will also be null when the client request times out. Currently this scenario also causes NPEs.

This PR addresses both scenarios by skipping filter calls when the filters are null. I've also added tests for both scenarios.

ganskef commented:
@jekh I see the motivation and I agree. But, consider to set a HttpFiltersAdapter (NullObject), if the FiltersSource returns null. I don't like the null checks and the risk to miss one.

jekh commented:
I agree, avoiding the null checks is a much better solution. I've updated the PR to avoid the null checks. Instead, the proxy will always use a default no-op HttpFiltersAdapter instance, unless the HttpFiltersSource overrides it with a non-null value.

I also included a minor modification to the way bad requests are written, introduced in PR #220, so the 400s provide a Content-Length and are handled properly in response to HEAD requests. I don't typically like to do two things in a PR, but for this minor change, this is the easiest way to avoid merge conflicts.

ganskef commented on src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java:
```diff
@@ -22,18 +22,17 @@
 import io.netty.handler.traffic.GlobalTrafficShapingHandler;
 import io.netty.util.concurrent.Future;
 import io.netty.util.concurrent.GenericFutureListener;
-
```
@jekh We should use common formatting rules, which are integrated with the build so it not depends the IDE.

jekh replied:
I agree, IntelliJ reformatted the includes automatically without me telling it to. It would be especially nice to get a default column width of 120 or 140. There's a lot of code that wraps way too early.

ganskef commented:
@jekh DirectRequestTest.testAnswerFromFilterShouldBeServed() passes after code is moved, so I have no reservations about this PR.

jekh commented:
Thanks, @ganskef. I'll go ahead an merge this, then.
